### PR TITLE
feat: add C implementation for `math/base/special/fast/abs`

### PR DESCRIPTION
**Pull request by gunjjoshi**
Resolves a part of #1917 .

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/fast/abs`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/fast/abs).

```C
double stdlib_base_fast_abs( const double x )
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #1917.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   ☑️ Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Pranavchiku** [comment]
Will have to discuss if this is needed and / or what can be suitable C API.

File: lib/node_modules/@stdlib/math/base/special/fast/abs/src/main.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/fast/abs.h"
+
+/**
+* Computes the absolute value of `x`.
+*
+* @param x    input value
+* @returns    absolute value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fast/abs/src/main.c]:
```suggestion
* @return     absolute value
```

**Comment by kgryte** [comment]
Adding a C implementation for the "fast" implementations is fine. Moving to "needs review".

File: lib/node_modules/@stdlib/math/base/special/fast/abs/lib/native.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* var v = abs( 2.0 );
+* // returns 2.0
+*
+* @example
+* var v = abs( 0.0 );
+* // returns 0.0
+*
+* @example
+* var v = abs( -0.0 );
+* // returns -0.0
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/abs/lib/native.js:45]:
```suggestion
* // returns 0.0
```

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/fast/abs/lib/native.js:45]: I'll commit it in batch, please wait.

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @gunjjoshi !

@gunjjoshi please open an issue tracking C implementation for `math/base/special/fast/*`

**Comment by gunjjoshi** [comment]
> LGTM. Thanks @gunjjoshi !
> 
> @gunjjoshi please open an issue tracking C implementation for `math/base/special/fast/*`

Sure, I'll do that.

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**